packages/app/cypress/e2e/runner/support/spec-loader.ts

```diff
@@ -78,3 +78,23 @@ export function loadSpec (options: LoadSpecOptions): void {
   // Wait for specs to complete
   shouldHaveTestResults({ passCount, failCount, pendingCount })
 }
+
+export function runSpec ({ fileName }: { fileName: string }) {
+  cy.scaffoldProject('runner-e2e-specs')
+  cy.openProject('runner-e2e-specs')
+  cy.startAppServer()
+
+  cy.__incorrectlyVisitAppWithIntercept()
+
+  cy.findByLabelText('Search Specs').type(fileName)
+  // wait for virtualized spec list to update, there is a chance
+  // of disconnection otherwise
+  cy.wait(500)
+  cy.contains('[data-cy=spec-item]', fileName).click()
```

Comment:
From what I can tell, #20558 corrected the issue with the runner hanging when directly visiting a spec. So you should be able to do something like:

```js
// we'd need the relative filePath rather than fileName ('runner/file.cy.ts' vs. 'file.cy.ts') to
// build the url correctly
export function runSpec ({ filePath }: { filePath: string }) {
  cy.scaffoldProject('runner-e2e-specs')
  cy.openProject('runner-e2e-specs')
  cy.startAppServer()

  // Need to normalize the spec path for windows usage
  cy.visitApp(getRunnerHref(`cypress/e2e/${filePath}`))

  // Assert runner has loaded before continuing, eventManager isn't initialized otherwise
  // There's probably a better way to assert this?
  cy.contains('.runnable-header', Cypress._.last(filePath.split('/')) as string).should('be.visible')

  return cy.window()
}
```

Doing so would shave quite a bit of time off those test runs. I'm actively working on updating the loadSpec function above.

Reply:
What does your `getRunnerHref` look like? I might just wait for your refactor to land and grab it (or if this merges first, make the change after).

Reply:
We have an [existing implementation](https://github.com/cypress-io/cypress/blob/10.0-release/packages/app/cypress/e2e/specs.cy.ts#L4-L10) that we could generalize, but let's wait on this. I'll make sure this gets updated along with the other implementation once this merges.